scripts/Crawlers/BaseCrawler.py

```diff
@@ -552,3 +558,22 @@ def _check_dats_present(self, directory):
                         return os.path.join(file_path, subfile_name)
             elif file_name.lower() == "dats.json":
                 return file_path
+
+    def _add_source_data_submodule_if_derivedfrom_conp_dataset(
+        self, dats_json, dataset_dir
+    ):
+        f = open(dats_json)
```

Comment:
I would suggest using the `with` context here. i.e. `with open(..., 'r') as f: ... `

Reply:
fixed in the next commit